src/System.Runtime/tests/System/DateTime.cs

```diff
@@ -291,6 +294,78 @@ public static void TestTryParseExactA()
         Assert.Equal(s, actual);
     }
 
+    [Fact]
+    [ActiveIssue(846, PlatformID.AnyUnix)]
+    public static void TestGetDateTimeFormats()
+    {
+        var savedCulture = CultureInfo.CurrentCulture;
+
+        try
+        {
+            CultureInfo.CurrentCulture = new CultureInfo("en-US");
+            DateTime july28 = new DateTime(2009, 7, 28, 5, 23, 15);
+
+            List<string> actualJuly28Formats = july28.GetDateTimeFormats().ToList();
+
+            Assert.Equal(expectedJuly28Formats.OrderBy(t => t), actualJuly28Formats.OrderBy(t => t));
+        }
+        finally
+        {
+            CultureInfo.CurrentCulture = savedCulture;
+        }
+    }
+
+    [Fact]
+    [ActiveIssue(846, PlatformID.AnyUnix)]
+    public static void TestGetDateTimeFormats_FormatSpecifier()
+    {
+        var savedCulture = CultureInfo.CurrentCulture;
+
+        try
+        {
+            char[] allStandardFormats =
+            {
+                'd', 'D', 'f', 'F', 'g', 'G',
+                'm', 'M', 'o', 'O', 'r', 'R',
+                's', 't', 'T', 'u', 'U', 'y', 'Y',
+            };
+
+            CultureInfo.CurrentCulture = new CultureInfo("en-US");
+            DateTime july28 = new DateTime(2009, 7, 28, 5, 23, 15);
+
+            List<string> actualJuly28Formats = new List<string>();
+            foreach (char format in allStandardFormats)
+            {
+                actualJuly28Formats.AddRange(july28.GetDateTimeFormats(format));
+            }
+
+            Assert.Equal(expectedJuly28Formats.OrderBy(t => t), actualJuly28Formats.OrderBy(t => t));
+        }
+        finally
+        {
+            CultureInfo.CurrentCulture = savedCulture;
+        }
+    }
+
+    [Fact]
+    public static void TestGetDateTimeFormats_FormatSpecifier_InvalidFormat()
+    {
+        DateTime july28 = new DateTime(2009, 7, 28, 5, 23, 15);
+
+        Assert.Throws<FormatException>(() => july28.GetDateTimeFormats('x'));
+    }
+
+    [Fact]
+    [ActiveIssue(846, PlatformID.AnyUnix)]
+    public static void TestGetDateTimeFormats_FormatProvider()
+    {
+        DateTime july28 = new DateTime(2009, 7, 28, 5, 23, 15);
+
+        List<string> actualJuly28Formats = july28.GetDateTimeFormats(new CultureInfo("en-US")).ToList();
```

Comment:
We cannot assume the date/time formats are not going to change. this wrong. we shouldn't assume any specific formats in our tests. if we need to go to the details, we should use our specific format in our test (like mm/dd/yyyy) and tests the results.